pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/CountAggregationFunction.java

```diff
@@ -50,7 +50,11 @@ public CountAggregationFunction(ExpressionContext expression) {
   public CountAggregationFunction(ExpressionContext expression, boolean nullHandlingEnabled) {
     super(expression);
     // Consider null values only when null handling is enabled and function is not COUNT(*)
-    _nullHandlingEnabled = nullHandlingEnabled && !expression.getIdentifier().equals("*");
+    // Note COUNT on any literal gives same result as COUNT(*)
+    // So allow for identifiers that are not * and functions, disable for literals and *
+    _nullHandlingEnabled = nullHandlingEnabled
+            && ((expression.getType() == ExpressionContext.Type.IDENTIFIER && !expression.getIdentifier().equals("*"))
+            || (expression.getType() == ExpressionContext.Type.FUNCTION));
```

Comment:
What's the motivation of adding function to enable null count? 

The behavior should be :
When nullHandling is enabled
count() should ignore all nulls exception it is count(*)

Reply:
Previous behaviour was count ignore all nulls for all expressions that are identifiers that are not *. Expressions that are not identifiers (literals or functions) result in exception because of the old check. 
New behaviour is ignore all nulls for all expressions that are not count (*) or count(any literal). count(*) and count(literal) both return rows. It misses count(null) case tho.